### DISCUSSION TITLE
A and B are changing

### DISCUSSION THREAD

**Discussion by Tianfneg** [TinyMPC/TinyMPC#26] [General]
Thanks for the answer, I would like to understand that if the A and B matrices are varying with path curvature and velocity, can Pinf still be solved only once?

**Comment by plancherb1** [comment]
Unfortunately, if you have a varying A and B, you'd need to solve for a series of Ks and Ps. That said, if you have enough memory you could store a batch of them to use online.

**Reply by plancherb1** [reply]
We'd love to support that more general case in our codebase too so do submit a PR if you happen to implement that and we'd be very happy to support that option as well!

**Reply by Tianfneg** [reply]
Thanks a lot for your reply, I think it might be advantageous to be able to buffer some of the data based on interval data

**Reply by Tianfneg** [reply]
> 我们也很乐意在我们的代码库中支持这种更通用的情况，所以如果您碰巧实现了它，请提交 PR，我们也非常乐意支持该选项！
Could you please share the UAV state parameter model you used? Because I'm a bit confused as to whether the parameter model for the drone doesn't vary with trajectory or speed?

**Reply by xkhainguyen** [reply]
> > 我们也很乐意在我们的代码库中支持这种更通用的情况，所以如果您碰巧实现了它，请提交 PR，我们也非常乐意支持该选项！
> > Could you please share the UAV state parameter model you used? Because I'm a bit confused as to whether the parameter model for the drone doesn't vary with trajectory or speed?

Please check [https://github.com/TinyMPC/tinympc-julia/blob/main/examples/interactive_quadrotor_ext.ipynb](https://github.com/TinyMPC/tinympc-julia/blob/main/examples/interactive_quadrotor_ext.ipynb). We provide a detailed example of MPC on a quadrotor (real Crazyflie model that we've been using). Basically we linearize the nonlinear dynamics around the hovering point to obtain a LTI model with A and B matrices which remain unchanged for the entire program.

**Reply by Tianfneg** [reply]
Hello, can you give a short answer about the Xref_data data being in absolute coordinate system? I input the data in absolute coordinate system and change A,B to my vehicle model, but when tracking the y-direction the number is getting smaller and smaller almost to 0.

**Reply by xkhainguyen** [reply]
Could you describe your problem in detail in another post?

**Comment by Tianfneg** [comment]
Hello, the top three figures in each grid are the desired trajectories I was given, and the bottom three figures are the output from our tracking process. Is it because I entered the wrong type of reference trajectory?
![pig](https://github.com/TinyMPC/TinyMPC/assets/147623854/e76e6095-8924-45bb-8b92-1c800c0fc181)

**Reply by hzw4868** [reply]
大佬您好，请问能否分享一下您用tinyMPC做的车辆模型跟踪代码呢，感激不尽
